platform/linux-generic/odp_pool.c

```diff
@@ -1052,6 +1092,9 @@ int buffer_alloc_multi(pool_t *pool, odp_buffer_hdr_t *buf_hdr[], int max_num)
 					       burst);
 		cache_num = burst - num_deq;
 
+		if (CONFIG_POOL_STATISTICS && pool->params.stats_opt.bit.alloc_ops)
+			odp_atomic_inc_u64(&pool->stats.alloc_ops);
```

Comment:
Can ring_ptr_deq_multi() return zero, in which case alloc_ops probably shouldn't be incremented?

Reply:
True. I think API spec needs to address what is counted as an alloc operation.

Reply:
Updated API suggestion:
```
/**
 * Pool statistics counters options
 *
 * Pool statistics counters listed in a bit field structure.
 */
typedef union odp_pool_stats_opt_t {
	/** Option flags */
	struct {
		/** @see odp_pool_stats_t::available */
		uint64_t available          : 1;

		/** @see odp_pool_stats_t::alloc_ops */
		uint64_t alloc_ops          : 1;

		/** Alloc operations counter includes operations which didn't
		 *  return any events */
		uint64_t alloc_ops_pool_empty : 1;

		/** @see odp_pool_stats_t::free_ops */
		uint64_t free_ops           : 1;

		/** @see odp_pool_stats_t::combined_ops */
		uint64_t combined_ops       : 1;

		/** Combined operations counter includes operations which didn't
		 *  return any events */
		uint64_t combined_ops_empty : 1;

		/** @see odp_pool_stats_t::cache_available */
		uint64_t cache_available    : 1;

		/** @see odp_pool_stats_t::cache_alloc_ops */
		uint64_t cache_alloc_ops    : 1;

		/** @see odp_pool_stats_t::cache_free_ops */
		uint64_t cache_free_ops     : 1;

		/** @see odp_pool_stats_t::cache_combined_ops */
		uint64_t cache_combined_ops : 1;
	} bit;

	/** All bits of the bit field structure
	 *
	 *  This field can be used to set/clear all flags, or for bitwise
	 *  operations over the entire structure. */
	uint64_t all;

} odp_pool_stats_opt_t;

/**
 * Pool statistics counters
 */
typedef struct odp_pool_stats_t {
	/** The number of available entries in the pool */
	uint64_t available;

	/** The number of alloc operations from the pool */
	uint64_t alloc_ops;

	/** The number of free operations to the pool */
	uint64_t free_ops;

	/** The combined number of alloc and free operations */
	uint64_t combined_ops;

	/** The number of available entries in the local caches of all threads
	 *  using the pool */
	uint64_t cache_available;

	/** The number of alloc operations from pool caches, which returned
	 *  at least one event. */
	uint64_t cache_alloc_ops;

	/** The number of free operations, which stored events to pool caches. */
	uint64_t cache_free_ops;

	/** The combined number of alloc and free operations, which moved at
	 *  least one event. */
	uint64_t cache_combined_ops;

} odp_pool_stats_t;

```